pkg/events/derive/derive.go

```diff
@@ -72,3 +82,25 @@ func makeEventSkeleton(eventID events.ID) eventSkeleton {
 		Params: def.Params,
 	}
 }
+
+func getInvokingEventContext(event *trace.Event) (trace.Event, error) {
```

Comment:
Maybe something simpler like `withInvokingContext` similar to how golang contexts do it.